app/src/main/java/com/nearbynest/app/ui/list/PropertyListFragment.kt

```diff
@@ -15,14 +15,58 @@
  */
 package com.nearbynest.app.ui.list
 
+import androidx.appcompat.widget.SearchView
 import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.nearbynest.app.R
+import com.nearbynest.app.data.fakedata.FakeData
 import com.nearbynest.app.databinding.FragmentPropertyListBinding
 import com.nearbynest.app.ui.base.BaseFragment
+import com.nearbynest.app.utils.extension.observeEvent
+import dagger.hilt.android.AndroidEntryPoint
+import timber.log.Timber
 
+@AndroidEntryPoint
 class PropertyListFragment : BaseFragment<FragmentPropertyListBinding, PropertyListViewModel>() {
 
     override val viewModel: PropertyListViewModel by viewModels()
 
     override fun getLayoutResId(): Int = R.layout.fragment_property_list
+
+    private lateinit var propertyListAdapter: PropertyListAdapter
+
+    override fun initialize() {
+        super.initialize()
+        setupRecyclerView()
+        binding.searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String?): Boolean {
+                return false
+            }
+
+            override fun onQueryTextChange(newText: String?): Boolean {
+                if (newText.isNullOrEmpty()) return false
+                propertyListAdapter.filter.filter(newText)
+                return true
+            }
+        })
+        Timber.d(message = "${binding.rvPropertyList.adapter?.itemCount}")
+    }
+
+    private fun setupRecyclerView() {
+        binding.rvPropertyList.apply {
+            layoutManager = LinearLayoutManager(requireContext())
+            propertyListAdapter = PropertyListAdapter()
+            adapter = propertyListAdapter
+        }
+        viewModel.imageList.value?.let { propertyListAdapter.imageList = it.peekContent() }
+        propertyListAdapter.addAllItem(ArrayList(FakeData.propertyDetailsList))
+    }
+
+    override fun setupViewModel() {
+        super.setupViewModel()
+        viewModel.imageList.observeEvent(viewLifecycleOwner) {
+            propertyListAdapter.imageList = it
```

Comment:
You should never do this. Always feed data to the adapter and read from submitted data. You should be never having separate data sources for the adapter item.

Reply:
Changes done to model class.
Adapter will be having data with single source.